x/poolmanager/router.go

```diff
@@ -744,17 +743,12 @@ func (k Keeper) EstimateTradeBasedOnPriceImpactBalancerPool(
 	swapModule types.PoolModuleI,
 	poolI types.PoolI,
 ) (*queryproto.EstimateTradeBasedOnPriceImpactResponse, error) {
-	// There isn't a case where the tokenOut could be zero or an error is received but those possibilities are handled
-	// anyway.
 	tokenOut, err := swapModule.CalcOutAmtGivenIn(ctx, poolI, req.FromCoin, req.ToCoinDenom, sdk.ZeroDec())
 	if err != nil {
```

Comment:
Re: comment about removing the wrapper functions, instead of calling wrapper functions we can do something like this

```suggestion
    if err != nil {
        if errors.Is(err, gammtypes.ErrInvalidMathApprox) {
    }
```